### PR TITLE
Log per-disk errors during quorum errors

### LINKED ISSUE #5276: Improve error message for quorum errors
## Expected Behavior

When quorum errors happen, it would be useful to log the underlying per-disk error on the server that had a quorum error - this would reveal the underlying error and concerned disks more quickly.

## Current Behavior

Currently quorum errors do not provide any details about underlying per-disk errors.

## Context

In https://github.com/minio/minio/issues/5271 a quorum error was due to insufficient inodes on a particular disk but it was not apparent from console logs.

### PR DESCRIPTION
**Pull request by krisis**
## Description

This change prints per-disk errors when quorum is not met like,
```
      ...
      27: net/http/server.go:2619:http.serverHandler.ServeHTTP()
      28: net/http/server.go:1801:http.(*conn).serve()
[2018-01-31T09:37:31.213667493Z] [ERROR] 
	disk [1/4]: disk not found
	disk [2/4]: disk not found (Write failed. Insufficient number of disks online)
```
## Motivation and Context
Fixes #5276 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — COMMENTED** [review]
For logging changes would love @kannappanr @ebozduman to chime in here.

File: cmd/xl-v1-utils.go [hunk 1]
```diff
@@ -77,6 +79,18 @@ func reduceQuorumErrs(errs []error, ignoredErrs []error, quorum int, quorumErr e
...
 		// No quorum satisfied.
 		maxErr = errors2.Trace(quorumErr, errs...)
+
+		// Log errors from disks to help admin take action.
+		var diskErrLogs []interface{}
+		var errCount int
+		for i, diskErr := range errs {
+			if diskErr != nil {
+				diskErrLogs = append(diskErrLogs,
+					fmt.Sprintf("disk [%d/%d]: %v", i+1, len(errs), diskErr.Error()))
```
Comment (kannappanr) [inline cmd/xl-v1-utils.go:89]: As part of changes in https://github.com/minio/minio/pull/5450 , we have decided to use static strings and map to represent the values( in this case, the disk number and disk error).

Comment (krisis) [reply cmd/xl-v1-utils.go:89]: @kannappanr OK. In that case let's block this PR until error logging changes are merged.

**Comment by harshavardhana** [comment]
Since the logging has changed quite a bit - @krisis should this PR be closed?

**Comment by ebozduman** [comment]
@kp,
I agree with @harshavardhana. After logging refactoring task, this part of the code has changed dramatically.
Now it looks like below:
```
func reduceQuorumErrs(ctx context.Context, errs []error, ignoredErrs []error, quorum int, quorumErr error) error {
  maxCount, maxErr := reduceErrs(errs, ignoredErrs)
  if maxCount >= quorum {
    logger.LogIf(ctx, maxErr)
    return maxErr
  }
  logger.LogIf(ctx, quorumErr)
  return quorumErr
}
```
Hence this PR can be closed.

**Comment by harshavardhana** [comment]
Closing this @krisis feel free to re-open if you want to enhance or add more to the current logging.

**Pull request closed without merging**